File: MultipartMultipleImagePost/app/src/main/java/com/multipleimage/uploadfile/APIService.java
package com.multipleimage.uploadfile;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Header;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Part;

interface APIService {
    /*@Multipart
    @POST("/upload_multi_files/MultiUpload.php")
    Call<ResponseBody> uploadFile(@Part MultipartBody.Part file, @Part("name") RequestBody name);*/
    @Multipart
    @POST("/immigration/api/updateProfile")
    Call<Model> uploadSingleFile(
            @Header("accessToken") String accessToken,
            @Part MultipartBody.Part file,
            @Part("firstName") RequestBody firstName,
            @Part("lastName") RequestBody lastName,
            @Part("contact") RequestBody contact,
            @Part("countryCode") RequestBody countryCode);

    @Multipart
    //@POST("/upload_multi_files/MultiUpload.php")
    @POST("/upload_multi_files/MultiPartUpload.php")
    Call<Model> uploadMultiFile(@Part MultipartBody.Part file1, @Part MultipartBody.Part file2, @Part MultipartBody.Part file3);

    //@Multipart
    //@FormUrlEncoded
    //@POST("/upload_multi_files/MultiUpload.php")
    @POST("/ride")
    Call<ResponseBody> uploadMultiFile(
            @Header("access_token") String accessToken,
            @Body RequestBody file);

    @FormUrlEncoded
    @PUT("/api/register")
    Call<ResponseBody> getStatus(
            @Field("username") String username,
            @Field("email") String email,
            @Field("password") String password,
            @Field("fbID") String fbID,
            @Field("gmailID") String gmailID,
            @Field("twitID") String twitID,
            @Field("gender") String gender,
            @Field("birthDate") String birthDate,
            @Field("location") String location,
            @Field("longitude") String longitude,
            @Field("latitude") String latitude,
            @Field("profileImage") String profileImage);
    //@Field parameters can only be used with form encoding. (parameter #2)
}

File: RetrofitProfile/app/src/main/java/com/retrofitprofile/restservices/APIService.kt
package com.immigration.restservices


import com.retrofitdemo.retrofit.model.ResponseModel
import com.retrofitdemo.retrofit.restservices.Constant.key_Fname
import com.retrofitdemo.retrofit.restservices.Constant.key_Lname
import com.retrofitdemo.retrofit.restservices.Constant.key_accessToken
import com.retrofitdemo.retrofit.restservices.Constant.key_contact
import com.retrofitdemo.retrofit.restservices.Constant.key_countryCode
import com.retrofitdemo.retrofit.restservices.Constant.urlUpdateProfile
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*


interface APIService {
   
   
   
   @FormUrlEncoded
   @POST("/user/login")
   fun Login(@Field("phone_number") phone_number:String,
             @Field("password")password:String,
             @Field("country_code") country_code:String
             ):Call<ResponseModel>
   
   
   
   @Multipart
   @POST("/immigration/api/updateProfile")
   fun postImage(@Header(key_accessToken) accessToken:String,
                 @Part image: MultipartBody.Part?,
                 @Part(key_Fname) firstName: RequestBody,
                 @Part(key_Lname) lastName: RequestBody,
                 @Part(key_contact) contact: RequestBody,
                 @Part(key_countryCode) countryCode: RequestBody
   ): Call<ResponseModel>
   
   
}